### ISSUE TITLE
The "Bold text" setting does not work for some texts

### ISSUE THREAD

**Issue by irfano** [wordpress-mobile/WordPress-Android#18274]
### Expected behavior
All text should be bold when the "Bold text" setting is enabled. 

### Actual behavior
The "Bold text" setting doesn't make some texts bold. Additionally, other views may not support "Bold text" and should be investigated and fixed accordingly. If current behavior is expected, please close the issue.

<img width="1132" alt="Screenshot 2023-04-14 at 17 55 20" src="https://user-images.githubusercontent.com/2471769/232079364-830a3d69-cb5a-45ca-885c-462df3a9f16c.png">

### Steps to reproduce the behavior
1. Enable the "Bold text" from the "Display size and text" settings of the device. "Bold text" feature is only available for Android 12+.
2. Launch the app and log in.
3. Notice domains and blaze cards on the dashboard don't contain bold text. 
4. Navigate to "Me → About". Notice texts are not bold.

**Comment by ravishanker** [comment]
Fixed for dashboard card domain in https://github.com/wordpress-mobile/WordPress-Android/pull/18286

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by jkmassel** [comment]
There are probably wider accessibility issues around, but we can address those more specifically.
